src/libraries/System.Text.Json/gen/JsonSourceGenerator.Emitter.cs

```diff
@@ -315,70 +312,44 @@ private static string GenerateForTypeWithKnownConverter(TypeGenerationSpec typeM
                 string typeCompilableName = typeMetadata.TypeRef;
                 string typeFriendlyName = typeMetadata.TypeInfoPropertyName;
 
-                string metadataInitSource = $@"_{typeFriendlyName} = {JsonMetadataServicesTypeRef}.{GetCreateValueInfoMethodRef(typeCompilableName)}({OptionsInstanceVariableName}, {JsonMetadataServicesTypeRef}.{typeFriendlyName}Converter);";
+                string metadataInitSource = $@"ret = {JsonMetadataServicesTypeRef}.{GetCreateValueInfoMethodRef(typeCompilableName)}(options, {JsonMetadataServicesTypeRef}.{typeFriendlyName}Converter);";
```

Comment:
Why the change? The point behind using metavariables is to avoid emitting invalid code being emitted. How can you guarantee that the `ret` and `options` are in scope in methods that splice `metadataInitSource`?

Also, nit: given that generated code is meant to be read by users, `ret` is not a very descriptive name.

Reply:
good point, will change. Generally speaking since I generate the code I know what the variables are called - extra indirection doesn't make this easier to read IMO. `ret` perhaps could have better name though